### ISSUE TITLE
Impossible to use loose mode for classes

### ISSUE THREAD

**Issue by developit** [rollup/rollup-plugin-babel#47]
[L20](https://github.com/rollup/rollup-plugin-babel/blob/master/src/index.js#L20) makes it impossible to use the class transformer's loose specified by presets passed to `rollup-plugin-babel`. This is because the class transform is manually added to the plugins list, which takes precedence over preset-provided plugins.

Is this intentional?  If not, what about something like this?

```js
const classCheck = transform( 'class Foo {}', options ).code;
if (~classCheck.indexOf('class Foo')) {
    options.plugins.push( classes );
}
```

**Comment by Rich-Harris** [comment]
Hmm – no, it looks like the options should get cloned rather than mutated. Most likely an oversight on my part...

**Comment by developit** [comment]
@Rich-Harris If I manually specify `['transform-es2015-classes', { loose:true }]` in `options.plugins`, I get loose mode classes.  I'm only seeing the issue when a preset specifies loose mode, such as [this one](https://github.com/developit/babel-preset-es2015-minimal-rollup).

**Comment by eventualbuddha** [comment]
This appears to no longer exhibit the same problem, so I'm closing. Please comment if I have that wrong.

**Comment by eventualbuddha** [comment]
Actually, I think it [may still be a problem](https://github.com/rollup/rollup-plugin-babel/blob/3ebbaa1b1d6c1092b9f27102d28c762e880beabd/src/preflightCheck.js#L17).

**Comment by developit** [comment]
@eventualbuddha confirmed, still an issue.  Tested by removing [these transforms in preact-svg](https://github.com/developit/preact-svg/blob/master/rollup.config.js#L32-L33), which generates strict mode class output at `dist/preact-svg.js`.

**Comment by Rich-Harris** [comment]
Just released 2.5.1 which no longer mutates `plugins` – can you verify that it behaves correctly please? thanks

**Comment by developit** [comment]
@Rich-Harris Confirmed, it's fixed! Thanks muchly.
